### PR TITLE
chore: upgrade to node 20

### PR DESCRIPTION
**Pull request by gergelypap**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true
+    "source.fixAll.eslint": "explicit"
```
Comment (gergelypap) [inline .vscode/settings.json:3]: This was changed by VSCode automatically: https://stackoverflow.com/questions/77637621/vscode-workspace-settings-change-on-its-own

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -4,41 +4,40 @@ on: [pull_request]
...
+      - name: Link
+        run: npm link
+      - name: Run unit tests
+        run: npm test
+        env:
+          CI: true
+      - name: Run build
+        run: npm run build
+      - name: Run integration tests
+        run: node ./lib/index.js
```
Comment (csutorasr) [inline .github/workflows/ci.yaml:43]: Please keep it as it was.

File: .github/workflows/publish.yaml [hunk 3]
```diff
@@ -6,17 +6,16 @@ on:
...
+      - uses: actions/checkout@v3
+      # Setup .npmrc file to publish to npm
+      - uses: actions/setup-node@v3
+        with:
+          node-version: '20.x'
+          registry-url: 'https://registry.npmjs.org'
+      - run: npm install
+      - run: npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (csutorasr) [inline .github/workflows/publish.yaml:21]: Same for this.

**Review by csutorasr — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (gergelypap) [reply .github/workflows/ci.yaml]: done

**Pull request merged**